storage/innobase/trx/trx0rseg.cc

```diff
@@ -107,18 +127,15 @@ void trx_rseg_update_wsrep_checkpoint(const XID* xid)
 
 	trx_rseg_update_wsrep_checkpoint(rseg_header, xid, &mtr);
 
-	const byte* xid_uuid = wsrep_xid_uuid(xid);
-	if (memcmp(wsrep_uuid, xid_uuid, sizeof wsrep_uuid)) {
-		memcpy(wsrep_uuid, xid_uuid, sizeof wsrep_uuid);
-
+	if (must_clear_rsegs) {
 		/* Because the UUID part of the WSREP XID differed
 		from current_xid_uuid, the WSREP group UUID was
 		changed, and we must reset the XID in all rollback
 		segment headers. */
 		for (ulint rseg_id = 1; rseg_id < TRX_SYS_N_RSEGS; ++rseg_id) {
 			if (const trx_rseg_t* rseg =
 			    trx_sys.rseg_array[rseg_id]) {
-				trx_rseg_update_wsrep_checkpoint(
+				trx_rseg_write_wsrep_checkpoint(
```

Comment:
Should we call here `trx_rseg_clear_wsrep_checkpoint()` similar to 10.4 and implement `trx_rseg_clear_wsrep_checkpoint()` by writing null XID?  `mlog_memset()` which is used in 10.4 is not available in 10.3 to my knowledge.

Reply:
OK, done. Test running here: https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/708/

Reply:
Made one small change to trx_rseg_clear_wsrep_checkpoint(). Running tests again here https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/709/